hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -896,6 +907,40 @@ public void reclaimPreviouslyReservedThrottle(int n, HederaFunctionality functio
         synchronizedThrottleAccumulator.leakUnusedThrottlePreviouslyReserved(n, function);
     }
 
+    @Override
+    public boolean shouldThrottleNOfUnscaled(int n, HederaFunctionality function) {
+        return synchronizedThrottleAccumulator.shouldThrottleNOfUnscaled(n, function, userTransactionConsensusTime);
+    }
+
+    public boolean shouldThrottleTxn(TransactionInfo txInfo) {
+        return networkUtilizationManager.shouldThrottle(txInfo, current());
+    }
+
+    @Override
+    public boolean hasThrottleCapacityForChildTransactions() {
+        var isAllowed = true;
+        final var childRecords = recordListBuilder.childRecordBuilders();
+
+        for (int i = 0, n = childRecords.size(); i < n && isAllowed; i++) {
+            final var childRecord = childRecords.get(i);
+            if (Objects.equals(childRecord.status(), SUCCESS)) {
+                final var tx = childRecord.build().transaction();
+                final var txInfo = new TransactionInfo(
+                        tx, childRecord.transactionBody(), tx.sigMap(), tx.signedTransactionBytes(), functionality);
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**issue:** ‏I think `functionality` here is the one of the of the context, but the one we need should be derived from the transaction body of the child record probably using `functionOf(childRecord.transactionBody())`.